### ISSUE TITLE
I probably clearly have no idea what I'm doing. Forgive me. 

### ISSUE THREAD

**Issue by Sephiths** [mircokroon/minecraft-world-downloader#86]
PS C:\Users\(user)\Downloads> java -jar world-downloader.jar -s (server ip)
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: Launcher has been compiled by a more recent version of the Java Runtime (class file version 53.0), this version of the Java Runtime only recognizes class file versions up to 52.0
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(Unknown Source)
        at java.security.SecureClassLoader.defineClass(Unknown Source)
        at java.net.URLClassLoader.defineClass(Unknown Source)
        at java.net.URLClassLoader.access$100(Unknown Source)
        at java.net.URLClassLoader$1.run(Unknown Source)
        at java.net.URLClassLoader$1.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at sun.launcher.LauncherHelper.checkAndLoadMain(Unknown Source)

I checked to make sure my Java version was up to date. And I tried downloading earlier releases. 
Do I have to do something seperate to update "Java Runtime"?

**Comment by szalony9szymek** [comment]
uninstall your jre and than install [jre 10](https://www.oracle.com/java/technologies/java-archive-javase10-downloads.html) (you need a free acount to download it)

**Comment by szalony9szymek** [comment]
Java SE Runtime Environment 10.0.2 for your os from the page

**Comment by Sephiths** [comment]
C:\WINDOWS\system32>java -jar C:\Users\jgrif\Downloads\world-downloader.jar -s play.spinkour.com
java.nio.file.NoSuchFileException: world\region
        at java.base/sun.nio.fs.WindowsException.translateToIOException(Unknown Source)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
        at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(Unknown Source)
        at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(Unknown Source)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.readAttributes(Unknown Source)
        at java.base/java.nio.file.Files.readAttributes(Unknown Source)
        at java.base/java.nio.file.FileTreeWalker.getAttributes(Unknown Source)
        at java.base/java.nio.file.FileTreeWalker.visit(Unknown Source)
        at java.base/java.nio.file.FileTreeWalker.walk(Unknown Source)
        at java.base/java.nio.file.FileTreeIterator.<init>(Unknown Source)
        at java.base/java.nio.file.Files.walk(Unknown Source)
        at java.base/java.nio.file.Files.walk(Unknown Source)
        at game.data.WorldManager.getMcaFiles(WorldManager.java:122)
        at game.data.WorldManager.outlineExistingChunks(WorldManager.java:84)
        at gui.GuiManager.createAndShowGUI(GuiManager.java:56)
        at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.desktop/java.awt.EventQueue.access$600(Unknown Source)
        at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
        at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Reading profile information from C:\Users\jgrif\AppData\Roaming\.minecraft\launcher_profiles.json
Starting proxy for play.spinkour.com. Make sure to connect to localhost:25565 instead of the regular server address.
Performed handshake with play.spinkour.com:25565, protocol version 578 :: next state: login
Login by: Sephiths
java.lang.RuntimeException: Client not authenticated! {
  "error" : "ForbiddenOperationException",
  "path" : "/session/minecraft/join"
}
        at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:167)
        at proxy.EncryptionManager.lambda$sendReplacementEncryptionConfirmation$7(EncryptionManager.java:255)
        at proxy.EncryptionManager.disconnectOnError(EncryptionManager.java:98)
        at proxy.EncryptionManager.sendReplacementEncryptionConfirmation(EncryptionManager.java:255)
        at proxy.EncryptionManager.lambda$setClientEncryptionConfirmation$6(EncryptionManager.java:243)
        at proxy.EncryptionManager.attempt(EncryptionManager.java:85)
        at proxy.EncryptionManager.setClientEncryptionConfirmation(EncryptionManager.java:232)
        at packets.builder.ServerBoundLoginPacketBuilder.lambda$new$1(ServerBoundLoginPacketBuilder.java:28)
        at packets.builder.PacketBuilder.build(PacketBuilder.java:46)
        at packets.DataReader.readPackets(DataReader.java:161)
        at packets.DataReader.pushData(DataReader.java:116)
        at proxy.ProxyServer.lambda$run$4(ProxyServer.java:100)
        at proxy.ProxyServer.attempt(ProxyServer.java:165)
        at proxy.ProxyServer.lambda$run$6(ProxyServer.java:97)
        at java.base/java.lang.Thread.run(Unknown Source)
Client probably disconnected. Waiting for new connection...
Server probably disconnected. Waiting for new connection...


I've tried launching it before starting up the minecraft launcher
I've tried starting it while already in-game
I can connect to other servers just not through local host
I've tried making sure to connect to a non-bungee server. 

It seems to launch now atleast. But I still can't connect.

**Comment by Sephiths** [comment]
![image](https://user-images.githubusercontent.com/33009114/99961894-363b5a80-2d5d-11eb-88b1-b061d6849784.png)
This may be because I didn't quite download java from the proper source. 
But I didn't feel like waiting
![image](https://user-images.githubusercontent.com/33009114/99961942-4fdca200-2d5d-11eb-89eb-e970ab9f93f1.png)

**Comment by mircokroon** [comment]
Hmm the issues doesn't seem related to that, do you maybe have an empty `world/` folder in the same folder as the downloader? Try removing it if it's there, or otherwise changing the output location (with `-o output` for example).

**Comment by Sephiths** [comment]
Well. I went ahead and switched to java 15. Same error as far as I can tell though. So I assume it'll still work if I figure out how to do this right. 

Yep. The default launcher. Although it updated to prepare to support the microsoft logins recently. Maybe like a week ago? Are you sure it's working since then/with the latest launcher? 

I've used modded clients but that doesn't effect the launcher. Havent tried with the vanilla versions/clients. 

The program created an empty world folder. and I have tried deleting it and running it again.. Well, almost empty. It creates the DIM1, DOM-1, and region folder without any actual information inside. 

Okay trying your suggestions now.

**Comment by Sephiths** [comment]
okay so I deleted the world folder. As well as created an output folder to direct it to. No change. The "output" folder now stores the DIM1, DOM-1, and region folder

**Comment by Sephiths** [comment]
I used powershell this time as it seemed easier. This is the log. 


PS C:\Users\jgrif\Downloads> java -jar C:\Users\jgrif\Downloads\world-downloader.jar -s reborn.pw -o C:\Users\jgrif\Downloads\output
Reading profile information from C:\Users\jgrif\AppData\Roaming\.minecraft\launcher_profiles.json
Starting proxy for mc.reborn.pw. Make sure to connect to localhost:25565 instead of the regular server address.
Performed handshake with reborn.pw:25565, protocol version 578 :: next state: login
Login by: Sephiths
java.lang.RuntimeException: Client not authenticated! {
  "error" : "ForbiddenOperationException",
  "path" : "/session/minecraft/join"
}
        at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:167)
        at proxy.EncryptionManager.lambda$sendReplacementEncryptionConfirmation$7(EncryptionManager.java:255)
        at proxy.EncryptionManager.disconnectOnError(EncryptionManager.java:98)
        at proxy.EncryptionManager.sendReplacementEncryptionConfirmation(EncryptionManager.java:255)
        at proxy.EncryptionManager.lambda$setClientEncryptionConfirmation$6(EncryptionManager.java:243)
        at proxy.EncryptionManager.attempt(EncryptionManager.java:85)
        at proxy.EncryptionManager.setClientEncryptionConfirmation(EncryptionManager.java:232)
        at packets.builder.ServerBoundLoginPacketBuilder.lambda$new$1(ServerBoundLoginPacketBuilder.java:28)
        at packets.builder.PacketBuilder.build(PacketBuilder.java:46)
        at packets.DataReader.readPackets(DataReader.java:161)
        at packets.DataReader.pushData(DataReader.java:116)
        at proxy.ProxyServer.lambda$run$4(ProxyServer.java:100)
        at proxy.ProxyServer.attempt(ProxyServer.java:165)
        at proxy.ProxyServer.lambda$run$6(ProxyServer.java:97)
        at java.base/java.lang.Thread.run(Thread.java:832)
Client probably disconnected. Waiting for new connection...
Server probably disconnected. Waiting for new connection...

**Comment by mircokroon** [comment]
Right, I have not tried it since the launcher update, but I don't think they have made any breaking changes yet. I'll go give it a try.

**Comment by mircokroon** [comment]
Yeah it still works as before for me. I do see in your output that it says:
`Reading profile information from C:\Users\jgrif\AppData\Roaming.minecraft\launcher_profiles.json`

Somehow the `\` is missing between the roaming and .minecraft, or is that just due to copy-pasting?

**Comment by Sephiths** [comment]
C:\WINDOWS\system32>java -jar C:\Users\jgrif\Downloads\world-downloader.jar -s reborn.pw -o C:\Users\jgrif\Downloads\output -m C:\Users\jgrif\AppData\Roaming\.minecraft
Reading profile information from C:\Users\jgrif\AppData\Roaming\.minecraft\launcher_profiles.json
Starting proxy for mc.reborn.pw. Make sure to connect to localhost:25565 instead of the regular server address.
Performed handshake with reborn.pw:25565, protocol version 754 :: next state: login
Login by: Sephiths
java.lang.RuntimeException: Client not authenticated! {
  "error" : "ForbiddenOperationException",
  "path" : "/session/minecraft/join"
}
        at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:167)
        at proxy.EncryptionManager.lambda$sendReplacementEncryptionConfirmation$7(EncryptionManager.java:255)
        at proxy.EncryptionManager.disconnectOnError(EncryptionManager.java:98)
        at proxy.EncryptionManager.sendReplacementEncryptionConfirmation(EncryptionManager.java:255)
        at proxy.EncryptionManager.lambda$setClientEncryptionConfirmation$6(EncryptionManager.java:243)
        at proxy.EncryptionManager.attempt(EncryptionManager.java:85)
        at proxy.EncryptionManager.setClientEncryptionConfirmation(EncryptionManager.java:232)
        at packets.builder.ServerBoundLoginPacketBuilder.lambda$new$1(ServerBoundLoginPacketBuilder.java:28)
        at packets.builder.PacketBuilder.build(PacketBuilder.java:46)
        at packets.DataReader.readPackets(DataReader.java:161)
        at packets.DataReader.pushData(DataReader.java:116)
        at proxy.ProxyServer.lambda$run$4(ProxyServer.java:100)
        at proxy.ProxyServer.attempt(ProxyServer.java:165)
        at proxy.ProxyServer.lambda$run$6(ProxyServer.java:97)
        at java.base/java.lang.Thread.run(Thread.java:832)
Client probably disconnected. Waiting for new connection...
Server probably disconnected. Waiting for new connection...

No change

**Comment by mircokroon** [comment]
As I said, run with `-m manual` instead of the Minecraft path. It will ask you for your username and token.

**Comment by Sephiths** [comment]
Microsoft Windows [Version 10.0.18363.1198]
(c) 2019 Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>java -jar C:\Users\jgrif\Downloads\world-downloader.jar -s reborn.pw -o C:\Users\jgrif\Downloads\output -m manual
Reading profile information from manual\launcher_profiles.json
Cannot read or find launcher_profiles.json!
You can use launch option "-m /path/to/.minecraft/" to indicate the location of your Minecraft installation.
Using manual authentication...

Minecraft username: Sephiths
Found user 'Sephiths' with UUID 'a48405613f30440faac7b0e8769c019d'

Your access token is needed for authentication. It...
        - Can be found in launcher_profiles.json, inside your .minecraft directory (for the default launcher)
        - Should be named 'accessToken'
        - Is quite long (over 300 characters)
        - Can change when starting the game, so launch Minecraft before entering it
If you have multiple accounts, make sure to pick the one matching the given username.
Access token: ~removed~
Starting proxy for mc.reborn.pw. Make sure to connect to localhost:25565 instead of the regular server address.
Performed handshake with reborn.pw:25565, protocol version 754 :: next state: login
Login by: Sephiths
java.lang.RuntimeException: Client not authenticated! {
  "error" : "ForbiddenOperationException",
  "path" : "/session/minecraft/join"
}
        at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:167)
        at proxy.EncryptionManager.lambda$sendReplacementEncryptionConfirmation$7(EncryptionManager.java:255)
        at proxy.EncryptionManager.disconnectOnError(EncryptionManager.java:98)
        at proxy.EncryptionManager.sendReplacementEncryptionConfirmation(EncryptionManager.java:255)
        at proxy.EncryptionManager.lambda$setClientEncryptionConfirmation$6(EncryptionManager.java:243)
        at proxy.EncryptionManager.attempt(EncryptionManager.java:85)
        at proxy.EncryptionManager.setClientEncryptionConfirmation(EncryptionManager.java:232)
        at packets.builder.ServerBoundLoginPacketBuilder.lambda$new$1(ServerBoundLoginPacketBuilder.java:28)
        at packets.builder.PacketBuilder.build(PacketBuilder.java:46)
        at packets.DataReader.readPackets(DataReader.java:161)
        at packets.DataReader.pushData(DataReader.java:116)
        at proxy.ProxyServer.lambda$run$4(ProxyServer.java:100)
        at proxy.ProxyServer.attempt(ProxyServer.java:165)
        at proxy.ProxyServer.lambda$run$6(ProxyServer.java:97)
        at java.base/java.lang.Thread.run(Thread.java:832)
Client probably disconnected. Waiting for new connection...
WARNING: packet parsing may have been incorrect! Expected length: 261. Used bytes: 0
Server probably disconnected. Waiting for new connection...

I followed the instructions to find the access token. No luck.

**Comment by Sephiths** [comment]
To be more clear, I found the access token and connected. It just wasn't the solution*

**Comment by mircokroon** [comment]
You are able to connect without the downloader right? 

There seem to be [several potential causes](https://wiki.vg/Authentication#Errors) of this, not sure what could be the problem. Do you have multiple accounts in your launcher-profiles file? Perhaps you picked the wrong one?

**Comment by Sephiths** [comment]
Removed my extra accounts and tried again. Restarted MC and Launcher twice. 
Made sure to try manual and without. And also checked to make sure the access token matched my main account/the account I was logging in with. The launcher_profiles.json was also freshly generated/I had to refresh my text editor. 
No luck.

**Comment by mircokroon** [comment]
I've tried the same server and version and it works for me still, very strange. You could perhaps try running it from an administrator powershell, in case that makes a difference (really should not matter). 

What is the version number of your launcher? You can see it under settings -> about

**Comment by Sephiths** [comment]
![image](https://user-images.githubusercontent.com/33009114/99973802-2bd58c80-2d6e-11eb-85c2-54e63a6bdca6.png)

**Comment by mircokroon** [comment]
Ah I see, I guess the changes to the launcher are only in the _beta_ version which I wasn't using. I can reproduce the problem now, I'll go poke around for a solution.

**Comment by Sephiths** [comment]
Ah! That explains it! Thanks! 
I forgot I was using the Beta. So I guess I gave you an early warning that it's about to break? Good luck! 



So. It's certainly working better now. I'm still having issues though.

**Comment by Sephiths** [comment]
Microsoft Windows [Version 10.0.18363.1198]
(c) 2019 Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>java -jar C:\Users\jgrif\Downloads\world-downloader.jar -s play.spinkour.com -o C:\Users\jgrif\Downloads\output
Reading profile information from C:\Users\jgrif\AppData\Roaming\.minecraft\launcher_profiles.json
Starting proxy for play.spinkour.com. Make sure to connect to localhost:25565 instead of the regular server address.
Performed handshake with play.spinkour.com:25565, protocol version 578 :: next state: login
Login by: Sephiths
Successfully authenticated user with Mojang session server.
User identity confirmed with Mojang.
Enabled encryption
Enabled compression
Login success: Sephiths logged in with uuid a4840561-3f30-440f-aac7-b0e8769c019d
Using protocol of game version 1.15.0 (578)
Looks like we have not run in version 1.15.0 before.
Downloading this version's server.jar (https://launcher.mojang.com/v1/objects/e9f105b3c5c7e85c7b445249a93362a22f62442d/server.jar)
java.nio.file.NoSuchFileException: cache\server.jar
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
        at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
        at java.base/java.nio.file.Files.newOutputStream(Files.java:224)
        at java.base/java.nio.file.Files.write(Files.java:3491)
        at game.data.registries.RegistryLoader.downloadServerJar(RegistryLoader.java:105)
        at game.data.registries.RegistryLoader.getReportsFromServerJar(RegistryLoader.java:89)
        at game.data.registries.RegistryLoader.<init>(RegistryLoader.java:54)
        at game.Game.loadVersionRegistries(Game.java:166)
        at game.Game.lambda$getGameProtocol$0(Game.java:158)
        at java.base/java.lang.Thread.run(Thread.java:832)

**Comment by Sephiths** [comment]
I didn't mean to close oops. I got it to work with the server above. But I didn't save the log of it doing so. I haven't managed to get it to work on any other server, or even on that same server again.

**Comment by mircokroon** [comment]
Oh boy. Well I added some exception handling for that in the [latest version](https://github.com/mircokroon/minecraft-world-downloader/releases/tag/v1.8.2), hopefully it'll help. 

Apparently on the beta version of the launcher, it creates another file called `launcher_accounts.json` with the actual valid accessToken in there. I added some handling for that now, so if it exists it will try to use that token. Probably that means it won't work for you now unless you delete `launcher_accounts.json` or switch back to the beta launcher.

**Comment by Sephiths** [comment]
After going back to the beta to test. 
I still manage to join the server now. For about a second before it kicks me. Just like when I downgraded the launcher. 

It is an improvement. Beforehand on this launcher version it wouldn't even connect me. But I still can't stay long enough/download the world. 


`Microsoft Windows [Version 10.0.18363.1198]
(c) 2019 Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>java -jar C:\Users\jgrif\Downloads\world-downloader.jar -s play.spinkour.com -o C:\Users\jgrif\Downloads\output
Reading profile information from C:\Users\jgrif\AppData\Roaming\.minecraft\launcher_profiles.json
Reading account information from C:\Users\jgrif\AppData\Roaming\.minecraft\launcher_accounts.json
Starting proxy for play.spinkour.com. Make sure to connect to localhost:25565 instead of the regular server address.
Performed handshake with play.spinkour.com:25565, protocol version 578 :: next state: login
Login by: Sephiths
Disconnect: {"translate":"Connection throttled! Please wait before reconnecting."}
Server probably disconnected. Waiting for new connection...
Performed handshake with play.spinkour.com:25565, protocol version 578 :: next state: login
Login by: Sephiths
Using accessToken from launcher_accounts.json
Successfully authenticated user with Mojang session server.
User identity confirmed with Mojang.
Enabled encryption
Enabled compression
Login success: Sephiths logged in with uuid a4840561-3f30-440f-aac7-b0e8769c019d
Using protocol of game version 1.15.0 (578)
Looks like we have not run in version 1.15.0 before.
Downloading this version's server.jar (https://launcher.mojang.com/v1/objects/e9f105b3c5c7e85c7b445249a93362a22f62442d/server.jar)
java.nio.file.NoSuchFileException: cache\server.jar
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
        at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
        at java.base/java.nio.file.Files.newOutputStream(Files.java:224)
        at java.base/java.nio.file.Files.write(Files.java:3491)
        at game.data.registries.RegistryLoader.downloadServerJar(RegistryLoader.java:111)
        at game.data.registries.RegistryLoader.getReportsFromServerJar(RegistryLoader.java:89)
        at game.data.registries.RegistryLoader.<init>(RegistryLoader.java:54)
        at game.Game.loadVersionRegistries(Game.java:166)
        at game.Game.lambda$getGameProtocol$0(Game.java:158)
        at java.base/java.lang.Thread.run(Thread.java:832)

C:\WINDOWS\system32>`

**Comment by mircokroon** [comment]
Ah hold on, you're running in the `C:\WINDOWS\system32` directory now, so it'll try to write some Minecraft stuff there, which it's (hopefully) not allowed to do.

Try navigation to the directory the downloader is in first, as you were doing earlier on in this issue.

**Comment by Sephiths** [comment]
Yep! thanks! 
I switched to using cmd instead of powershell after awhile. As I feared I should be in cmd instead (Powershell happens to be a lot easier to open up inside any folder). I was mainly assuming I just had to make sure I was running the right file, which I did by listing it's full location. Noticing I was managing to actually run the program. I kept with it. 

Thanks so much for all the help!
